server/server.go

```
@@ -0,0 +1,168 @@
+			IsFolder: decoded.IsFolder,
+		},
+	}, nil
+}
+
+func (s *Server) Read(ctx context.Context, req *proto.ReadReq) (*proto.FileRes, error) {
+	owner := req.GetOwner();
+	oid, err := primitive.ObjectIDFromHex(req.GetId())
+	if err != nil {
+		return nil, fmt.Errorf("InvalidArgument", fmt.Sprintf("Could not convert to ObjectId: %v", err))
+	}
+
+	decoded, err := fileModel.FinedOne(mongoManager.Client, bson.M {mongoID : oid} )
+	
+	if strings.Compare(decoded.Owner, owner) == 0 {
+		return nil, fmt.Errorf("Permition", fmt.Sprintf("Permition denaid %s: %v", req.GetOwner(), err));
+	}
+
+	return &proto.FileRes{
+		File: &proto.File{
+			Id:       decoded.ID.Hex(),
+			Owner:    decoded.Owner,
+			Path:     decoded.Path,
+			Name:     decoded.Name,
+			IsFolder: decoded.IsFolder,
+		},
+	}, nil
+}
+
+
+func (s *Server) Update(ctx context.Context, req *proto.UpdateReq) (*proto.FileRes, error) {
+	owner := req.GetOwner();
+	oid, err := primitive.ObjectIDFromHex(req.GetFile().Id)
+	if err != nil {
+		return nil, fmt.Errorf("InvalidArgument", fmt.Sprintf("Could not convert to ObjectId: %v", err))
+	}
+
+	owner, name, path, isFolder := req.GetFile().Owner, req.GetFile().Name, req.GetFile().Path, req.GetFile().IsFolder;
+	file := fileModel.New(owner, name, path, isFolder);
+
+	decoded, err := file.Update(mongoManager.Client, bson.M{mongoID : oid})
+	if err != nil {
+		return nil, err;
+	}
+
+	return &proto.FileRes{
+		File: &proto.File{
+			Id:       decoded.ID.Hex(),
+			Owner:    decoded.Owner,
+			Path:     decoded.Path,
+			Name:     decoded.Name,
+			IsFolder: decoded.IsFolder,
+		},
+	}, nil
+}
+
+func (s *Server) Delete(ctx context.Context, req *proto.DeleteReq) (*proto.BoolRes, error) {
+	owner := req.GetOwner();
+	oid, err := primitive.ObjectIDFromHex(req.GetId())
+	if err != nil {
+		return nil, fmt.Errorf("InvalidArgument", fmt.Sprintf("Could not convert to ObjectId: %v", err))
+	}
+
+	decoded, err := fileModel.FinedOne(mongoManager.Client, bson.M {mongoID : oid} )
```

You ignored this error.

---

And rename to `file`.